pkg/gps/GpsMessage.php

```diff
@@ -169,8 +176,13 @@ public function getNativeMessage(): ?GoogleMessage
         return $this->nativeMessage;
     }
 
-    public function setNativeMessage(GoogleMessage $message = null): void
+    public function setNativeMessage(?GoogleMessage $message = null): void
     {
         $this->nativeMessage = $message;
     }
+
+    public function getAttributes(): array
```

Comment:
I don't see a `setAttributes()` setter method. I'd assume that something someone might want to do.

In your current code the only way to manipulate the attributes is to pass them via constructor, as headers (`$headers['attributes']`). They cannot be changed afterwards, and they are only "moved" to `attributes` property (and `unset` from `$headers` array) in constructor.

I'd suggest removing the changes from the constructor, and expect users to pass attributes using `setAttributes()` method only. This way there will be no "hidden" functionality in headers and there will be no discrepancy between `setHeaders()` and constructor.

WDYT?

Reply:
I do like this to make it usable without completion on the lib we use ( https://github.com/sroze/messenger-enqueue-transport).

Reply:
@Steveb-p i check and don't understand what is the `$properties` in `GPSMessage`. 
What this should be in ?

Reply:
I wasn't using that library, I don't know what it represents too :sweat_smile: 

Anyway, I understand your reasoning - why you put attributes as a property of headers array. You're trying to make it interoperable between packages.

To make it consistent I'd suggest:
1. Making `setHeaders` work in the exact same way, i.e. unpack the `$headers['attributes']` into a dedicated property. You could actually just call this method from the constructor, so you won't duplicate the code/behavior.
2. Add the explicit `setAttributes()` method regardless, for those that don't want the interoperability aspect and need that explicit call.

Reply:
I change to add attributes instead of using a key in header